store/cardStore.ts

```diff
@@ -0,0 +1,180 @@
+import { defineStore } from "pinia";
+import type { Field } from "~/types/models/Card";
+
+const generateUniqueId = () => {
+  return Math.floor(Math.random() * 90) + 10;
+};
+
+export interface CardFields {
+  label: string;
+  image: {
+    type: string;
+    image: File;
+  };
+  fields: {
+    id?: number;
+    type: string;
+    title?: string;
+    value: string;
+    category: string;
+  }[];
+}
+
+export const useCardStore = defineStore("card", () => {
+  const state = reactive({
+    currentCard: [] as Field[],
+    currentId: undefined as number | undefined,
+    currentCategory: '',
+    currentFieldId: undefined as number | undefined,
+    currentLabel: '',
+    isLoading: true,
+    isEditing: false,
+    modalTitle: '',
+    isModalOpen: false,
+    selectedColor: '',
+  });
+
+  const initialState = {
+    label: null as CardFields["label"] | null,
+    imageFields: [] as CardFields["image"][],
+    personalFields: [] as CardFields["fields"][],
+    generalFields: [] as CardFields["fields"][],
+    socialFields: [] as CardFields["fields"][],
+    messagingFields: [] as CardFields["fields"][],
+    businessFields: [] as CardFields["fields"][],
+  };
+
+  const fields = {
+    label: ref(initialState.label),
+    imageFields: ref(initialState.imageFields),
+    personalFields: ref(initialState.personalFields),
+    generalFields: ref(initialState.generalFields),
+    socialFields: ref(initialState.socialFields),
+    messagingFields: ref(initialState.messagingFields),
+    businessFields: ref(initialState.businessFields),
```

Comment:
why we still have groupings of fields here ?

Reply:
To separate data po ng mga fields.  but puwide lang pala gamitin yung category to group.

Reply:
Already revised